### PR TITLE
enable test_invalid_names test in rpc_test

### PR DESCRIPTION
**Pull request by mrshenli**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#28376 enable test_invalid_names test in rpc_test**

Differential Revision: [D18045158](https://our.internmc.facebook.com/intern/diff/D18045158)

### CODE REVIEW

**Comment by mrshenli** [comment]
Closing this PR as it is squashed into the previous one.

File: test/rpc_test.py [hunk 1]
```diff
@@ -291,11 +293,24 @@ def test_invalid_names(self):
...
                 self_name="".join(["a" for _ in range(500)]),
-                backend=TEST_CONFIG.rpc_backend,
-                self_rank=self.rank,
-                init_method=self.init_method,
             )
+
+        from torch.distributed.rpc.api import _agent
+        self.assertEqual(_agent, None)
+        # join_rpc() should not do anything as _agent is None
         rpc.join_rpc()
```
Comment (pritamdamania87) [inline test/rpc_test.py]: Do we need the join_rpc here if it doesn't do anything?

**Review by pritamdamania87 — APPROVED** [review]

File: test/rpc_test.py (hunk 1 above)
Comment (mrshenli) [reply test/rpc_test.py]: no, it's not necessary, just to make sure it's fine to call join_rpc() even if init failed.

**Comment by mrshenli** [comment]
Test failures are irrelevant, landing.

```
(core dumped) "$PYTHON" -m pytest -x -v --disable-warnings --junit-xml="$pytest_reports_dir/result.xml" --ignore "$caffe2_pypath/python/test/executor_test.py" --ignore "$caffe2_pypath/python/operator_test/matmul_op_test.py" --ignore "$caffe2_pypath/python/operator_test/pack_ops_test.py" --ignore "$caffe2_pypath/python/mkl/mkl_sbn_speed_test.py" ${rocm_ignore_test[@]} "$caffe2_pypath/python" "${EXTRA_TESTS[@]}"
Exited with code 134
```

```
20:56:56 ======================================================================
20:56:56 ERROR: test_download_url_to_file (__main__.TestHub)
20:56:56 ----------------------------------------------------------------------
20:56:56 Traceback (most recent call last):
20:56:56   File "C:\Jenkins\Miniconda3\lib\urllib\request.py", line 1318, in do_open
20:56:56     encode_chunked=req.has_header('Transfer-encoding'))
20:56:56   File "C:\Jenkins\Miniconda3\lib\http\client.py", line 1239, in request
20:56:56     self._send_request(method, url, body, headers, encode_chunked)
20:56:56   File "C:\Jenkins\Miniconda3\lib\http\client.py", line 1285, in _send_request
20:56:56     self.endheaders(body, encode_chunked=encode_chunked)
20:56:56   File "C:\Jenkins\Miniconda3\lib\http\client.py", line 1234, in endheaders
20:56:56     self._send_output(message_body, encode_chunked=encode_chunked)
20:56:56   File "C:\Jenkins\Miniconda3\lib\http\client.py", line 1026, in _send_output
20:56:56     self.send(msg)
20:56:56   File "C:\Jenkins\Miniconda3\lib\http\client.py", line 964, in send
20:56:56     self.connect()
20:56:56   File "C:\Jenkins\Miniconda3\lib\http\client.py", line 1400, in connect
20:56:56     server_hostname=server_hostname)
20:56:56   File "C:\Jenkins\Miniconda3\lib\ssl.py", line 407, in wrap_socket
20:56:56     _context=self, _session=session)
20:56:56   File "C:\Jenkins\Miniconda3\lib\ssl.py", line 817, in __init__
20:56:56     self.do_handshake()
20:56:56   File "C:\Jenkins\Miniconda3\lib\ssl.py", line 1077, in do_handshake
20:56:56     self._sslobj.do_handshake()
20:56:56   File "C:\Jenkins\Miniconda3\lib\ssl.py", line 689, in do_handshake
20:56:56     self._sslobj.do_handshake()
20:56:56 ConnectionResetError: [WinError 10054] An existing connection was forcibly closed by the remote host
```

**Pull request closed without merging**